### ISSUE TITLE
False positives for numbers within strings in number-*

### ISSUE THREAD

**Issue by tinovyatkin** [stylelint/stylelint#1898]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug with erroneously recognising leading zero inside variable value definition.

> Which rule, if any, is this issue related to?

`number-leading-zero` set to `never`

> What CSS is needed to reproduce this issue?

`$flagsImagePath: "https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/9.0.9/img/";`

> What stylelint configuration is needed to reproduce this issue?

```json
 {
    "extends": "stylelint-config-standard",
    "plugins": [
        "stylelint-scss"
    ],
    "rules": {
        "number-leading-zero": "never"
    }
  }
```

> Which version of stylelint are you using?

`7.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"VSCode plugin"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to SCSS variables."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

```
[stylelint] unexpected-trailing-zero
[stylelint] unexpected-leading-zero
```

**Comment by evilebottnawi** [comment]
@tinovyatkin Thanks, confirm

**Comment by davidtheclark** [comment]
Is this related only to SCSS? Or will it also happen with this?

```css
a::before {
  content: "https://cdnjs.cloudflare.com/ajax/libs/intl-tel-input/9.0.9/img/";
}
```

**Comment by jeddy3** [comment]
> Or will it also happen with this?

Yes, apparently so. I've updated the issue title and labels accordingly.

`number-no-trailing-zeros`, as well as `number-leading-zero`, also erroneously checks "numbers" within strings.

Would adapting these rules to use [style-search](https://github.com/davidtheclark/style-search#strings) help here?  Or is this not the right tool to reach for in this instance?

**Comment by davidtheclark** [comment]
> Would adapting these rules to use style-search help here?

Absolutely.

**Comment by jeddy3** [comment]
@tinovyatkin Would you like to [contribute a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs)?

**Comment by gaidarenko** [comment]
Don't you mind if I fix this bug?

**Comment by evilebottnawi** [comment]
@gaidarenko let's go
